modules/ding_provider/ding_provider.module

```diff
@@ -626,22 +632,45 @@ function ding_provider_build_entity_id($ids) {
       $options['collectionType'] = 'manifestation';
       $res = ting_do_search($query, 1, count($chunk), $options);
 
+      // Sometimes the well will be unreachable or not even one of the ids in
+      // chunk was searchable. In any case, this protects against a notice.
+      if (empty($res)) {
+        continue;
+      }
+
       foreach ($res->collections as $pid => $collection) {
         $faust = explode(':', $pid)[1];
-        if (isset($ids[$faust])) {
-          $ids[$faust] = $pid;
+        if ($key = array_search($faust, $ids)) {
+          $translated_ids[$faust] = $pid;
+          unset($ids[$key]);
         }
       }
     }
 
-    return $ids;
+    // The faust left in ids array where not searchable and we could therefore
+    // not determine the proper namespace. Since it's possible to load objects
+    // separately, that are not searchable with the library's agencyid, the best
+    // we can do here is to use the standard DBC agency id, which is used for
+    // basis-records. The providers can then try ting_get_object with this PID,
+    // which may work (if it's not an inter-library loan of a local post).
+    // TODO: We should provide a separate mechanism for translating faust
+    // without holdingsitem.agencyid filter. This way the providers will know
+    // which materials can be searched with the current agency id and generate
+    // a link to them.
```

Comment:
:+1: I appreciate the level of commenting here when dealing with a tricky situation.